Druid/DRPragmaBasedTypeSystem.class.st

```diff
@@ -29,3 +32,35 @@ DRPragmaBasedTypeSystem >> receiverTypes: aDRMessageSend [
 			  ifFalse: [
 			  DRClassType for: (self class environment at: typeName) ] ]
 ]
+
+{ #category : #accessing }
+DRPragmaBasedTypeSystem >> slangTyper [
+	
+	slang ifNil: [ | codeGenerator |
+		codeGenerator := MLVMCCodeGenerator new.
+		codeGenerator vmMaker: VMMaker new.
+		codeGenerator vmMaker vmmakerConfiguration: VMMakerConfiguration.
+		codeGenerator vmClass: CoInterpreter.
+		codeGenerator addClass: StackInterpreter.
+		codeGenerator addClass: CoInterpreter.
+		slang := codeGenerator.
+	].
+	^ slang
+]
+
+{ #category : #private }
+DRPragmaBasedTypeSystem >> typeOf: aNode [
+
+	^ aNode acceptVisitor: self
+]
+
+{ #category : #visiting }
+DRPragmaBasedTypeSystem >> visitInstanceVariableNode: aRBVariableNode [ 
+	
+	| slangType isPointer |
+	slangType := self slangTyper typeOfVariable: aRBVariableNode name.
```

Comment:
You don't need the method here? How does it know which variable is?

Reply:
easy, it's an instance variable node :)

Reply:
The node yes, but the `slangTyper` only receives the name 🤔